app/lib/dartdoc/backend.dart

```diff
@@ -320,3 +320,58 @@ class DartdocBackend {
     await deletePool.close();
   }
 }
+
+class ArchiveTracker {
+  final _pvMap = <String, _TrackedEntry>{};
+  final _sampleLines = <String>[];
+
+  ArchiveTracker() {
+    _logger.info('Tracking: init.');
+    new Timer.periodic(const Duration(minutes: 10), (_) => _cleanup());
+    new Timer.periodic(const Duration(minutes: 45), (_) => _logSample());
+  }
+
+  List<String> get samples => _sampleLines;
+
+  void track(DartdocEntry entry) {
+    final key = '${entry.packageName}/${entry.packageVersion}';
+    final tracked = _pvMap.putIfAbsent(key, () => new _TrackedEntry());
+    tracked.archiveSize = entry.archiveSize ?? 0;
+    tracked.totalSize = entry.totalSize ?? 0;
+    tracked.lastAccess = new DateTime.now();
+  }
+
+  void _cleanup() {
+    final now = new DateTime.now();
+    _pvMap.removeWhere((key, value) {
+      final age = value.lastAccess.difference(now).abs();
+      return age.inMinutes > 90;
+    });
+  }
+
+  void _logSample() {
+    // roughly one and half days of samples
+    if (_sampleLines.length > 48) {
+      _sampleLines.removeAt(0);
```

Comment:
Suggestion: This is an inefficient ring buffer, but since it's bounded to 48 entries every 45 minutes, this is fine. I suggest moving this code down to just before the new sample line is inserted.